tests/src/test/java/alluxio/server/ft/journal/raft/EmbeddedJournalIntegrationTest.java

```diff
@@ -149,6 +162,120 @@ public void resizeCluster() throws Exception {
     mCluster.notifySuccess();
   }
 
+  @Test
+  public void growCluster() throws Exception {
+    mCluster = MultiProcessCluster.newBuilder(PortCoordination.EMBEDDED_JOURNAL_GROW)
+        .setClusterName("EmbeddedJournalAddMaster").setNumMasters(2).setNumWorkers(0)
+        .addProperty(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.EMBEDDED.toString())
+        .addProperty(PropertyKey.MASTER_JOURNAL_FLUSH_TIMEOUT_MS, "5min")
+        // To make the test run faster.
+        .addProperty(PropertyKey.MASTER_EMBEDDED_JOURNAL_ELECTION_TIMEOUT, "750ms")
+        .addProperty(PropertyKey.MASTER_EMBEDDED_JOURNAL_HEARTBEAT_INTERVAL, "250ms").build();
+    mCluster.start();
+
+    AlluxioURI testDir = new AlluxioURI("/test");
+    FileSystem fs = mCluster.getFileSystemClient();
+    fs.createDirectory(testDir);
+    Assert.assertTrue(fs.exists(testDir));
+
+    // Validate current quorum size.
+    Assert.assertEquals(2,
+        mCluster.getJournalMasterClientForMaster().getQuorumInfo().getServerInfoList().size());
+
+    // Create and start a new master to join to existing cluster.
+    // Get new master address.
+    MasterNetAddress newMasterAddress = new MasterNetAddress(
```

Comment:
Is it possible to implement the create new master logics in the MultiProcessCluster and expose related methods? Since embedded journal is the default journal type now, we may use the create new master logics in more tests.

Reply:
I tried it and it's basically requires a complete rewrite of `MultiProcessCluster`.
This PR hack it by making it accept external address, so it's aware of external masters.